File: leetcode/problems/remove-element/v1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int index = 0;
        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != val) {
                nums[index++] = nums[j];
            }
        }
        return index;
    }
}

public class v1 {
    public static void main(String[] args) {
        int[] problem = {0, 2, 1, 2, 3, 2};
        Solution solution = new Solution();
        int ans = solution.removeElement(problem, 2);
        System.out.println(ans);
    }
}

/**
 * javac v1.java
 * java v1
 * >> Hello World!!
 * (or just use code runner and play button)
 */

File: leetcode/problems/remove-element/v1.rs
pub struct Solution;

impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut index = 0;
        for j in 0..nums.len() {
            if nums[j] != val {
                nums[index] = nums[j];
                index += 1
            }
        }
        return index as i32;
    }
}

fn main() {
    let mut nums = vec![0, 1, 2, 3, 2, 4, 2, 5];
    let ans = Solution::remove_element(&mut nums, 2);
    println!("{}", ans);
}
